source/slang/slang-ir-inline.cpp

```diff
@@ -492,4 +556,44 @@ void performMandatoryEarlyInlining(IRModule* module)
     pass.considerAllCallSites();
 }
 
+
+    // Defined in slang-ir-specialize-resource.cpp
+bool isResourceType(IRType* type);
+
+    /// An inlining pass that inlines calls functions that returns resources.
+    /// This is needed for glsl targets.
+struct GLSLResourceReturnFunctionInliningPass : InliningPassBase
+{
+    typedef InliningPassBase Super;
+
+    GLSLResourceReturnFunctionInliningPass(IRModule* module)
+        : Super(module)
+    {}
+
+    bool shouldInline(CallSiteInfo const& info)
+    {
+        if (isResourceType(info.callee->getResultType()))
+        {
+            return true;
+        }
```

Comment:
This probably makes it okay: if we are only inlining functions that we *had* to inline anyway, then it is better to inline and risk producing a function we can't emit to GLSL rather than not inline and have an invalid function with a resource `return`.

Reply:
I'm still surprised by this situation, though, because we have an existing pass that is supposed to be optimizing these cases. I am left wondering why the existing pass that was supposed to specialize resources in function outputs isn't applying for these ones.

Reply:
I need to go double check our code, but could it be that the `specializeResourceOutputs` pass only look at "trivial" functions? If I have a function that returns a resouce type, and that there is an unrelated `if` statement in the function, will that pass still work?